### PR TITLE
Feat: integrate extra 'team project' validation for concept endpoints

### PR DESCRIPTION
**Pull request by pieterlukasse**
Jira Ticket: [VADC-618](https://ctds-planx.atlassian.net/browse/VADC-618)

### Breaking Changes
- Endpoints that query cohorts now have an extra authorization check for "team projects", meaning the user needs specific Arborist permissions before querying a cohort

### Dependency updates
- Depends on more Arborist roles now

[VADC-618]: https://ctds-planx.atlassian.net/browse/VADC-618?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: middlewares/auth.go [hunk 1]
```diff
@@ -71,3 +74,33 @@ func PrepareNewArboristRequest(ctx *gin.Context, arboristEndpoint string) (*http
 	req.Header.Set("Authorization", authorization)
 	return req, nil
 }
+
+func HasAccessToAtLeastOne(teamProjects []string) bool {
+	// TODO - query Arborist
```
Comment (pieterlukasse) [inline middlewares/auth.go:79]: TODO

File: tests/controllers_tests/controllers_test.go [hunk 2]
```diff
@@ -127,7 +135,17 @@ func (h dummyCohortDefinitionDataModel) GetAllCohortDefinitions() ([]*models.Coh
...
 
-var conceptController = controllers.NewConceptController(*new(dummyConceptDataModel), *new(dummyCohortDefinitionDataModel))
+type dummyTeamProjectAuthz struct{}
+
+func (h dummyTeamProjectAuthz) TeamProjectValidationForCohort(ctx *gin.Context, cohortDefinitionId int) bool {
+	return true
+}
+
+func (h dummyTeamProjectAuthz) TeamProjectValidation(ctx *gin.Context, cohortDefinitionId int, filterCohortPairs []utils.CustomDichotomousVariableDef) bool {
+	return true
```
Comment (pieterlukasse) [inline tests/controllers_tests/controllers_test.go:145]: add `false` test

File: utils/parsing.go [hunk 3]
```diff
@@ -275,3 +275,28 @@ func ParseSourceIdAndCohortIdAndVariablesAsSingleList(c *gin.Context) (int, int,
...
+	}
+	return uniqueList
+}
+
+func GetUniqueCohortDefinitionIdsListFromRequest(cohortDefinitionId int, filterCohortPairs []CustomDichotomousVariableDef) []int {
+	var idsList []int
+	idsList = append(idsList, cohortDefinitionId)
+	if len(filterCohortPairs) > 0 {
+		for _, filterCohortPair := range filterCohortPairs {
+			idsList = append(idsList, filterCohortPair.CohortId1, filterCohortPair.CohortId2) // TODO - rename CohortId1/2 here to cohortDefinition1/2
```
Comment (tianj7) [inline utils/parsing.go:297]: Think this TODO can be cleared before pr is merged?

Comment (pieterlukasse) [reply utils/parsing.go:297]: yes, let me do that

Comment (pieterlukasse) [reply utils/parsing.go]: done

**Review by tianj7 — APPROVED** [review]
lgtm

**Pull request merged**